semantics/cpp/language/common/dynamic.k

```diff
@@ -576,26 +602,20 @@ module CPP-DYNAMIC-OTHER
      rule isEvalVal(_) => false [owise]
 
      // references aren't kresults because they need to get transformed into the object they refer to.
-     rule isKResult(Lbl:KLabel(_::CPPValue, _::Trace, T::CPPType)) => notBool isCPPRefType(T)
-          requires isValKLabel(#klabel(Lbl))
+     rule isKResult(lv(_, _, T::CPPType) #Or xv(_, _, T::CPPType) #Or prv(_, _, T::CPPType)) => notBool isCPPRefType(T)
 
-     rule Lbl:KLabel((loc(Base::SymBase, 0, _) => loc(Base, 0)), _::Trace, cppRefType)
-          requires isValKLabel(#klabel(Lbl))
+     rule lv((loc(Base::SymBase, 0, _) => loc(Base, 0)), _::Trace, cppRefType)
+     rule xv((loc(Base::SymBase, 0, _) => loc(Base, 0)), _::Trace, cppRefType)
 
      // not all references are lvalues. Some are xvalues. The individual
      // cases where this is true must handle this explicitly.
-     rule <k> Lbl:KLabel(loc(Base::SymBase, Offset::Int), Tr::Trace, cppRefType #as T::CPPType) => Lbl(V, Tr, innerType(T)) ...</k>
+     rule <k> lv(loc(Base::SymBase, Offset::Int), Tr::Trace, cppRefType #as T::CPPType) => lv(V, Tr, innerType(T)) ...</k>
+          <references>... loc(Base, Offset) |-> V::SymLoc ...</references>
+     rule <k> xv(loc(Base::SymBase, Offset::Int), Tr::Trace, cppRefType #as T::CPPType) => lv(V, Tr, innerType(T)) ...</k>
```

Comment:
This changes semantics: `xv` now rewrites to `lv`.

Reply:
Good catch!